packages/extension-ui/src/Popup/ImportLedger.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019-2021 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+import { ActionContext, Address, Button, ButtonArea, Dropdown, VerticalSpace, Warning } from '../components';
+import { useLedger } from '../hooks/useLedger';
+import useTranslation from '../hooks/useTranslation';
+import { createAccountHardware } from '../messaging';
+import { Header } from '../partials';
+import { ThemeProps } from '../types';
+import ledgerChains from '../util/legerChains';
+
+interface AccOption {
+  text: string;
+  value: number;
+}
+
+interface NetworkOption {
+  text: string;
+  value: string;
+}
+
+const AVAIL: number[] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19];
+const KUSAMA_GENESIS: string = ledgerChains[0].genesisHash[0];
+
+interface Props extends ThemeProps {
+  className?: string;
+}
+
+function ImportLedger ({ className }: Props): React.ReactElement {
+  const { t } = useTranslation();
+  const { getLedger } = useLedger();
+  const [address, setAddress] = useState<string | null>(null);
+  const [accountIndex, setAccountIndex] = useState<number>(0);
+  const [addressOffset, setAddressOffset] = useState<number>(0);
+  const [error, setError] = useState<string | null>(null);
+  const [isBusy, setIsBusy] = useState(false);
+  const [genesis, setGenesis] = useState(KUSAMA_GENESIS);
+  const onAction = useContext(ActionContext);
+  const name = useMemo(() => `ledger ${accountIndex}/${addressOffset}`, [accountIndex, addressOffset]);
+
+  useEffect(() => {
+    setIsBusy(true);
+
+    // I don't get how this is possible, but sometimes addIndex was a string...
+    getLedger(genesis).getAddress(false, Number(accountIndex), Number(addressOffset))
+      .then(({ address }) => {
+        setIsBusy(false);
+        setAddress(address);
+      }).catch((e: Error) => {
+        setIsBusy(false);
+        const errorMessage = e.message.includes('unkown')
+          ? t<string>(
+            'Is your ledger locked? Error from ledger: {{errorMessage}}',
```

Comment:
Ahhhh... ok, this is a simple solution for the apps UI - if Ledger, get the address first to ensure it is connected. then sign.